protocol/src/main/java/org/mifos/chatbot/protocol/facebookMessenger/FacebookMessengerChatService.java

```diff
@@ -0,0 +1,275 @@
+        try {
+            this.messenger.verifyWebhook(mode, verifyToken);
+            return ResponseEntity.ok(challenge);
+        } catch (MessengerVerificationException e) {
+            log.warn("Webhook verification failed: " + e.getMessage());
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).body(e.getMessage());
+        }
+    }
+
+    @RequestMapping(value = {"/", ""}, method = RequestMethod.POST)
+    public ResponseEntity<Void> handleCallback(@RequestBody final String payload, @RequestHeader(SIGNATURE_HEADER_NAME) final String signature) {
+        try {
+            this.messenger.onReceiveEvents(payload, of(signature), event -> {
+                if (event.isTextMessageEvent()) {
+                    handleTextMessageEvent(event.asTextMessageEvent());
+                } else if (event.isAttachmentMessageEvent()) {
+                    handleAttachmentMessageEvent(event.asAttachmentMessageEvent());
+                } else if (event.isQuickReplyMessageEvent()) {
+                    handleQuickReplyMessageEvent(event.asQuickReplyMessageEvent());
+                } else if (event.isPostbackEvent()) {
+                    handlePostbackEvent(event.asPostbackEvent());
+                } else if (event.isAccountLinkingEvent()) {
+                    handleAccountLinkingEvent(event.asAccountLinkingEvent());
+                } else if (event.isOptInEvent()) {
+                    handleOptInEvent(event.asOptInEvent());
+                } else if (event.isMessageEchoEvent()) {
+                    handleMessageEchoEvent(event.asMessageEchoEvent());
+                } else if (event.isMessageDeliveredEvent()) {
+                    handleMessageDeliveredEvent(event.asMessageDeliveredEvent());
+                } else if (event.isMessageReadEvent()) {
+                    handleMessageReadEvent(event.asMessageReadEvent());
+                } else {
+                    handleFallbackEvent(event);
+                }
+            });
+            return ResponseEntity.status(HttpStatus.OK).build();
+        } catch (MessengerVerificationException e) {
+            log.error(e.getStackTrace().toString());
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+        } catch (NotImplementedException e) {
+            log.error(e.getStackTrace().toString());
+            return ResponseEntity.status(HttpStatus.OK).build();
+        }
+    }
+
+    private boolean authUser(String username, String password) {
+        HttpClient client = HttpClientBuilder.create().build();
+        HttpPost post = new HttpPost(settings.getApiUrl() + "/authentication?username=" + username + "&password=" + password + "&tenantIdentifier=mifos");
+        try {
+            HttpResponse response = client.execute(post);
+            if(response.getStatusLine().getStatusCode() == 200) {
+                return true;
+            }
+        } catch (Exception e) {
```

Comment:
Let's catch the specific exception we're looking for and expect only, and not the general Exception.

Reply:
Changed it to IOException, as can be thrown by HttpResponse execute method